src/code/code_80140E80.c

```diff
@@ -1,21 +1,359 @@
+/**
+ * Set up a BG from a specified source image and draw it to the specified color image with func_80140EAC(), using the
+ * BG's settings. Position uses the default (0,0), and no rescaling is done even 1-cycle mode is enabled.
+ *
+ * @see func_80141008() for arguments.
+ */
+void func_8014116C(Gfx** gfxP, void* source, void* img, s32 width, s32 height, s32 cycleFlag) {
+    func_80141008(gfxP, source, img, width, height, 0.0f, 0.0f, SCALE_UNCHANGED, SCALE_UNCHANGED, cycleFlag);
+}
+
+// wrapper for func_80141008 with general arguments, used in func_80141200
+/**
+ * Set up a BG from a specified source image and draw it to the specified color image with func_80140EAC(). Fully
+ * general settings are available.
+ *
+ * @see func_80141008() for arguments.
+ */
+void func_801411B4(Gfx** gfxP, void* source, void* img, s32 width, s32 height, f32 x, f32 y, f32 scaleX, f32 scaleY,
+                   s32 cycleFlag) {
+    func_80141008(gfxP, source, img, width, height, x, y, scaleX, scaleY, cycleFlag);
+}
+
+// TODO work out what to do with this.
+#define gDPSetPrimColor_u32(pkt, m, l, d)                                                      \
+    _DW({                                                                                      \
+        Gfx* _g = (Gfx*)(pkt);                                                                 \
+                                                                                               \
+        _g->words.w0 = (_SHIFTL(G_SETPRIMCOLOR, 24, 8) | _SHIFTL(m, 8, 8) | _SHIFTL(l, 0, 8)); \
+        _g->words.w1 = (unsigned int)(d);                                                      \
+    })
+
+#define gDPSetLODColor(pkt, c, m, l, d)                                          \
+    _DW({                                                                        \
+        Gfx* _g = (Gfx*)(pkt);                                                   \
+                                                                                 \
+        _g->words.w0 = _SHIFTL(c, 24, 8) | _SHIFTL(m, 8, 8) | _SHIFTL(l, 0, 8)); \
+        _g->words.w1 = (unsigned int)(d);                                        \
+    })
+
+// internal, used in func_8014151C
+/**
+ * Most general redrawing function.
+ *
+ *  - Copies `source` to `img`,
+ *  - fills the current framebuffer with `this->primColor`,
+ *  - applies colour interpolation to `img`,
+ *  - copies `img` back to `source`, with rescaling.
+ *
+ * @param[in]     this
+ * @param[in,out] gfxP   Pointer to current displaylist
+ * @param[in,out] source Pointer to beginning of in/out color image
+ * @param[out]    img    Pointer to beginning of destination color image
+ * @param[in]     width  Output width in pixels
+ * @param[in]     height Output height in pixels
+ */
+void func_80141200(Struct_80140E80* this, Gfx** gfxP, void* source, void* img, s32 width, s32 height) {
+    Gfx* gfx = *gfxP;
+    s32 pad[3];
+
+    gDPPipeSync(gfx++);
+    // copy source to img
```

Comment:
What's the difference between `image` and `texture` in this context?

Reply:
See above about color images. For example, the z buffer is actually a color image, but you wouldn't treat it as a texture... usually.